DiscordBot.Commands/AddFactCommand.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Discord.WebSocket;
+using Discord;
+using DiscordBot.Commands.Core.CommandAttributes;
+using DiscordBot.Commands.Core;
+using DiscordBot.Core.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace DiscordBot.Commands
+{
+    [Name("addfact")]
+    [Description("Adds new random fact, single from text or all from attached txt file")]
+    [Option("text", "Add single random fact", CommandOptionType.String, false)]
+    [Option("file", "Add from file all random fact (each in new line)", CommandOptionType.Attachment, false)]
+    [RequiredPermission(GuildPermission.ManageGuild)]
+    public class AddFactCommand : Command
+    {
+        private readonly ILogger<AddResponseCommand> _logger;
+        private IRandomFactProvider _randomFactProvider;
+
+        public AddFactCommand(ILogger<AddResponseCommand> logger, IRandomFactProvider randomFactProvider)
+        {
+            _logger = logger;
+            _randomFactProvider = randomFactProvider;
+        }
+
+        public override async Task ExecuteAsync(SocketSlashCommand command)
+        {
+            try
+            {
+                var guildId = command.GuildId;
+                var optionText = command.GetOptionValue("text") as string;
+                var optionFile = command.GetOptionValue("file") as Attachment;
```

Comment:
I have added `GetOptionalValue<T>(string)` some time ago. Let's use that whenever possible.